### PR TITLE
feat: return browser_url for kafka admin UI

### PR DESCRIPTION
**Pull request by JasonPowr**
## Description
Managed Kafka Control plane API should return URL to the data plane UI for Kafka in order to improve user experience.
Users should be able to navigate from various integrations like CLI, Ansible directly to UI where they can see metrics/monitoring etc.

jira: https://issues.redhat.com/browse/MGDSTRM-5706

## Verification Steps
1. Create Kafka instance.
2. In the response body there should be a url that links to admin UI

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] ~~Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by pawelpaszki** [comment]
@JasonPowr - is this still a WiP? I just ran it and it works as expected

**Comment by JasonPowr** [comment]
> @JasonPowr - is this still a WiP? I just ran it and it works as expected

@pawelpaszki I just need to add some changes to tests and it should be OK.

**Comment by JasonPowr** [comment]
> @JasonPowr - is this still a WiP? I just ran it and it works as expected

@pawelpaszki should be good now. :)

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (machi1990) [inline internal/kafka/internal/api/public/api/openapi.yaml:1788]: I am wondering if there is a better way of sending these UI metadata back. 
/cc @miguelsorianod @pb82 this is like the discussion we had in the last backlog refinement about the content schema

**Comment by JameelB** [comment]
@JasonPowr can you rebase this PR when you get a chance please :)

File: openapi/kas-fleet-manager.yaml
Comment (machi1990) [inline openapi/kas-fleet-manager.yaml:892]: We should also bump the version https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/a8cbfdd30d7c48eceb33f68a625e4980577aaa08/openapi/kas-fleet-manager.yaml#L5 to 1.4.0

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (JameelB) [reply internal/kafka/internal/api/public/api/openapi.yaml:1788]: any follow up on this @machi1990? I wonder if we need to discuss this further before considering merging this PR?

File: internal/kafka/internal/config/kafka.go
Comment (JameelB) [inline internal/kafka/internal/config/kafka.go:42]: I wonder if we can use `http://localhost:8080/` as default? I believe the [UI](https://github.com/bf2fc6cc711aee1a0c2a/kafka-ui) can be run locally.

File: internal/kafka/internal/presenters/kafka.go
Comment (JameelB) [inline internal/kafka/internal/presenters/kafka.go:56]: I think this bit will never change: `application-services/streams/kafkas/`. We can move this to the saas template instead. 

```suggestion
		BrowserUrl:              fmt.Sprintf("%s%s/dashboard", validateBrowserUrl(browserUrl), reference.Id),
```

Comment (JameelB) [inline internal/kafka/internal/presenters/kafka.go:56]: Wdyt of adding the `/` after the `%s`? It may make it easier to read. 
```suggestion
		BrowserUrl:              fmt.Sprintf("%s/application-services/streams/kafkas/%s/dashboard", validateBrowserUrl(browserUrl), reference.Id),
```

In this case, you don't need to check if it has "/" at the end. It can be removed instead, if the browser url was defined with it. 

There's a function called [strings.TrimSuffix()](https://pkg.go.dev/strings#TrimPrefix) that you can use for this.

File: internal/kafka/internal/config/kafka.go
Comment (JameelB) [reply internal/kafka/internal/config/kafka.go:42]: @machi1990 have you done this by any chance? :thinking:

File: internal/kafka/internal/presenters/kafka.go
Comment (JasonPowr) [reply internal/kafka/internal/presenters/kafka.go:56]: I agree it would make it easier to read. :)

File: internal/kafka/internal/config/kafka.go
Comment (machi1990) [reply internal/kafka/internal/config/kafka.go:42]: No, this is not something I’ve done but I am sure that’s how the UI team is doing their devs. Maybe @wtrocki would have some input

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (machi1990) [reply internal/kafka/internal/api/public/api/openapi.yaml:1788]: There has not been followups on this. 
In my opinion, the answer to the comment I raised above about the other issue maybe applicable here.

File: internal/kafka/internal/presenters/kafka.go
Comment (JasonPowr) [reply internal/kafka/internal/presenters/kafka.go:56]: @JameelB Could I just remove it from the default values and add it here then remove the validation on it altogether ?

**Review by wtrocki — APPROVED** [review]

File: internal/kafka/internal/presenters/kafka.go
Comment (JameelB) [inline internal/kafka/internal/presenters/kafka.go:56]: don't need to check if it has a "/" suffix here, the `TrimSuffix()` will return the same string if it doesn't. 

```suggestion
		BrowserUrl:              fmt.Sprintf("%s/%s/dashboard", strings.TrimSuffix(browserUrl), reference.Id),
```

File: internal/kafka/internal/presenters/kafka.go
Comment (JameelB) [inline internal/kafka/internal/presenters/kafka.go:66]: We can remove this, calling `TrimSuffix()` should be enough :)

File: internal/kafka/internal/cmd/kafka/list.go
Comment (JameelB) [inline internal/kafka/internal/cmd/kafka/list.go:53]: Was testing this and found that the kafkaConfig type needs to be a pointer otherwise it can't find it in the di container. 

Also found that `MustResolveAll` can take multiple args so we don't need to call it twice :) 

```suggestion
	var kafkaService services.KafkaService
	var kafkaConfig *config.KafkaConfig
	env.MustResolveAll(&kafkaService, &kafkaConfig)
```

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (wtrocki) [inline internal/kafka/internal/api/public/api/openapi.yaml:1109]: Why this ?

Comment (JasonPowr) [reply internal/kafka/internal/api/public/api/openapi.yaml:1109]: I believe it was because of this discussion https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/838#discussion_r826056113

Comment (wtrocki) [reply internal/kafka/internal/api/public/api/openapi.yaml:1109]: Yes! Good suggestion by @JameelB 
Using default value for development purposes is ok, but OpenAPI is user facing so if possible can we leave examples and others with still console.redhat.com while default in the code will be localhost (and we can override that in deployment)

Comment (wtrocki) [reply internal/kafka/internal/api/public/api/openapi.yaml:1109]: This is just suggestion. Ignore me completely if that doesn't make sense :D

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (machi1990) [reply internal/kafka/internal/api/public/api/openapi.yaml:1788]: Resolving this comment in favor of letting the discussion to be progressed on the other issue.

File: internal/kafka/internal/api/public/api/openapi.yaml
Comment (JameelB) [inline internal/kafka/internal/api/public/api/openapi.yaml:1109]: Changed in wrong file. This one is generated by `make openapi/generate`, it should be changed in the openapi/kas-fleet-manager.yaml instead.

File: openapi/kas-fleet-manager.yaml
Comment (JameelB) [inline openapi/kas-fleet-manager.yaml:1423]: When changing this, let's add in the full url since it's an example of the response.  

```suggestion
        browser_url: "https://console.redhat.com/application-services/streams/kafkas/1isy6rq3jki8q0otmjqfd3ocfrg/dashboard"
```

**Review by JameelB — APPROVED** [review]
Looks good to me, verified locally and works as expected :+1: 

Would be great to also have @wtrocki approval here too :)

File: templates/service-template.yml
Comment (wtrocki) [inline templates/service-template.yml:450]: Description or value might be invalid here

**Comment by wtrocki** [comment]
This looks amazing! 
@JasonPowr World class job!

File: internal/kafka/internal/cmd/kafka/list.go
Comment (JasonPowr) [reply internal/kafka/internal/cmd/kafka/list.go:52]: Sorry for the late reply @ziccardi yes that is correct.

Comment (JameelB) [reply internal/kafka/internal/cmd/kafka/list.go:52]: @ziccardi looks like it has to be a pointer to a pointer. Was getting this error otherwise: 
```
service container di failure: /home/jbriones/projects/gitlab.com/service-delivery/managed-services-api/pkg/environments/environment.go:223: type config.KafkaConfig not exists in the container
```

It's because the `NewKafkaConfig` returns a pointer, so di can't find it unless it's that type. We do a similar thing here: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/integration/data_plane_cluster_test.go#L960

**Comment by JasonPowr** [comment]
> Reverified with latest changes and its working as expected :)
> 
> Can you regenerate the openapi spec again? I'm getting a different bindata.go than the one in this PR.

Done now :) @JameelB

**Pull request merged**
